### PR TITLE
pwall submit

### PR DESCRIPTION
**Pull request by ltiber**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: org.pwall.Pwall.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+    "app-id": "org.pwall.Pwall",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "40",
```
Comment (hfiguiere) [inline org.pwall.Pwall.json:4]: GOME 40 has bee EOL for over 18month.

44 is the current one.

File: org.pwall.Pwall.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+		    "install -D res/pwall/videoIcon.png /app/share/videoIcon.png",
+			"install -D res/pwall/clearbackup.sh /app/share/clearbackup.sh",
+		    "make -f makefile-flatpak",
+			"install -D pwall /app/bin/pwall",
+			"install -D flatpak/jhead /app/bin/jhead"
+            ],
+          "sources": [
+              {
+              "type": "git",
+              "url": "https://github.com/ltiber/pwall.git"
```
Comment (hfiguiere) [inline org.pwall.Pwall.json:27]: a commit or tag is required,

File: org.pwall.Pwall.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+             }    
+          ]
+      }
+  ],
+
+    "finish-args": [
```
Comment (hfiguiere) [inline org.pwall.Pwall.json:33]: finish args is usually found above `modules`.

File: org.pwall.Pwall.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+     "--socket=x11",
+     "--share=network",
+     "--filesystem=host",
```
Comment (hfiguiere) [inline org.pwall.Pwall.json:36]: this is a very broad filesystem permission

File: org.pwall.Pwall.json [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+              {
+              "type": "git",
+              "url": "https://github.com/ltiber/pwall.git"
+             }    
+          ]
+      }
+  ],
+
+    "finish-args": [
+     "--socket=x11",
```
Comment (hfiguiere) [inline org.pwall.Pwall.json:34]: `--share=ipc` is necessary for x11

File: org.pwall.Pwall.json [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+      "install -D flatpak/org.pwall.Pwall.appdata.xml /app/share/metainfo/org.pwall.Pwall.appdata.xml",
+			"install -D res/pwall/pwall.png /app/share/icons/hicolor/128x128/apps/org.pwall.Pwall.png",
+		    "install -D res/pwall/help.html /app/share/help.html",
+		    "install -D res/pwall/noimage.png /app/share/noimage.png",
+			"install -D res/pwall/novideo.png /app/share/novideo.png",
+		    "install -D res/pwall/videoIcon.png /app/share/videoIcon.png",
+			"install -D res/pwall/clearbackup.sh /app/share/clearbackup.sh",
+		    "make -f makefile-flatpak",
+			"install -D pwall /app/bin/pwall",
+			"install -D flatpak/jhead /app/bin/jhead"
```
Comment (hfiguiere) [inline org.pwall.Pwall.json:22]: this looks like a binary. How does that work to support both x86_64 an aarch64?

File: org.pwall.Pwall.json [hunk 7]
```diff
@@ -0,0 +1,39 @@
+{
+    "app-id": "org.pwall.Pwall",
```
Comment (hfiguiere) [inline org.pwall.Pwall.json:2]: is `pwall.org` under the control of the developers? It doesn't look like it has anything.

File: org.pwall.Pwall.json [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "40",
+    "sdk": "org.gnome.Sdk",
+    "command": "pwall",
+    "modules": [
+      {
+          "name": "pwall",
+          "buildsystem": "simple",
+          "build-commands": [
+			"install -D flatpak/pwall-flatpak.desktop /app/share/applications/org.pwall.Pwall.desktop",
```
Comment (hfiguiere) [inline org.pwall.Pwall.json:12]: why would this file be flatpak specific?

**Review by hfiguiere — CHANGES_REQUESTED** [review]
Some change are required to build.

Also indentation is all over the place making it harder to read.

File: org.pwall.Pwall.json (hunk 7 above)
Comment (ltiber) [reply org.pwall.Pwall.json:2]: I've no domain registered for this software. The idea was to make a correct id in 3 part. For the moment the domain is available...

Comment (hfiguiere) [reply org.pwall.Pwall.json:2]: Actually it's a requirement. You can use github if you don't have a domain.

https://docs.flatpak.org/en/latest/conventions.html?highlight=app-id#application-ids

File: org.pwall.Pwall.json (hunk 8 above)
Comment (hfiguiere) [reply org.pwall.Pwall.json:12]: you can use `destkop-file-edit` at install to change that.

File: org.pwall.Pwall.json (hunk 1 above)
Comment (ltiber) [reply org.pwall.Pwall.json:4]: I'll change the version.

File: org.pwall.Pwall.json (hunk 4 above)
Comment (ltiber) [reply org.pwall.Pwall.json:36]: the main function of pwall is to be a photos and videos library. I need to have access to this file wherever they are.

File: org.pwall.Pwall.json (hunk 5 above)
Comment (ltiber) [reply org.pwall.Pwall.json:34]: ok I'll add this property.

File: org.pwall.Pwall.json (hunk 7 above)
Comment (ltiber) [reply org.pwall.Pwall.json:2]: As you said, I can use github, I would think to use com.github.ltiber.Pwall 
But the documentation says it's better to use something like io.github.pwall.Pwall
but I'm not registered in github.io
any suggestion ?

**Comment by ltiber** [comment]
I've just commit a new manifest. Let me know if it's ok for you.

File: io.github.pwall.Pwall.json [hunk 9]
```diff
@@ -0,0 +1,54 @@
+{
+    "app-id":"io.github.pwall.Pwall",
+    "runtime":"org.gnome.Platform",
```
Comment (hfiguiere) [inline io.github.pwall.Pwall.json:3]: do you really need the gnome platform?

Comment (ltiber) [reply io.github.pwall.Pwall.json:3]: Yes my application uses gtk library.

Comment (hfiguiere) [reply io.github.pwall.Pwall.json:3]: gtk 3 is in freedesktop-sdk.

File: io.github.pwall.Pwall.json [hunk 10]
```diff
@@ -0,0 +1,54 @@
+{
+    "app-id":"io.github.pwall.Pwall",
```
[the remark this answers is not in the document]
Comment (hfiguiere) [reply io.github.pwall.Pwall.json:2]: it should be `io.github.ltiber.Pwall` otherwise

[the remark this answers is not in the document]
Comment (ltiber) [reply io.github.pwall.Pwall.json:2]: There was probably a misunderstanding.
I've suggested to use com.github.ltiber.Pwall as my url is https://github.com/ltiber/pwall 
but the doc said to use io pattern so I suggested to use io.github.pwall.Pwall 
But actually I think the app id should be io.github.ltiber.Pwall.

**Comment by ltiber** [comment]
I've just commit a new manifest with the appId io.github.ltiber.Pwall and using only freedesktop. I didn't know it came with gtk3 support. Thanks for the review.

**Comment by hfiguiere** [comment]
bot, build io.github.ltiber.Pwall

**Comment by ltiber** [comment]
I've just installed and test pwall from the testing repository and everything works fine.

**Pull request closed without merging**